server.py

```diff
@@ -72,6 +88,25 @@ def parse_request(request):
         return meth[1]
 
 
+def resolve_uri(parse):
+    root = os.path.join(os.getcwd(), 'webroot')
+    body = ''
+    content_type = ''
+    if os.path.isdir(root + parse):
+        body = '<!DOCTYPE html><html><body><ul>'
+        for file_ in os.listdir(root + parse):
+            body += '<li>' + file_ + '</li>'
+        body += '</ul></body></html>'
+        content_type = 'text/html'
+    elif os.path.isfile(root + parse):
+        with open((root + parse), 'rb') as file_:
+            body = file_.read()
+        content_type, encoding = mimetypes.guess_type(parse)
+    else:
+        raise OSError
```

Comment:
If this were an `IOError` instead, then it would synchronize nicely with the `IOError`s that might be raised by the calls to `open` and `os.listdir` above.  That would simplify exception handling outside calls to this function.